drivers/scheduler/helmchart/factory.go

```diff
@@ -0,0 +1,98 @@
+package helmchart
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// Factory is an application chart factory
+type Factory struct {
+	chartDir    string
+	chartParser Parser
+}
+
+var appChartFactory = make(map[string]*AppChart)
+
+// register registers a new chart with the factory
+func (f *Factory) register(id string, app *AppChart) {
+	if _, ok := appChartFactory[id]; !ok {
+		logrus.Infof("Registering new app: %v", id)
+	} else {
+		logrus.Infof("Substitute with new app: %v", id)
+	}
+	// NOTE: In case of chart rescan we need to substitute old app with another one
+	appChartFactory[id] = app
+}
+
+// Get returns a registered application
+func (f *Factory) Get(id string) (*AppChart, error) {
+	if d, ok := appChartFactory[id]; ok && d.Enabled {
+		if copy := d.DeepCopy(); copy != nil {
+			return d.DeepCopy(), nil
```

Comment:
Can we return `copy` obj here instead of again calling `d.DeepCopy
()`?

Reply:
Done